src/Writers/HTMLWriter.jl

```diff
@@ -74,67 +74,142 @@ const ASSETS_SASS = joinpath(ASSETS, "scss")
+- `class::Symbol`: Can be used to set the asset class, which determines how
+  exactly the asset gets included in the HTML page. This is necessary if the
+  class can not be determined automatically (default). Valid symbols are `:js`,
+  `:css` or `:ico`.
 
-**`islocal`** can be used to declare the asset to be local. The `uri` should then be a path
-relative to the documentation source directory (conventionally `src/`). This can be useful
-when it is necessary to override the asset class of a local asset.
+- `islocal::Bool`: Can be used to declare the asset to be local `islocal=true`.
+  The `src` should then be a path relative to the documentation source
+  directory (conventionally `src/`) or a absolute path pointing to a valid file.
+  This can be useful when it is necessary to override the asset class of a
+  local asset. If it is not manually set it will be set to `false` in case of
+  the `src` starting with `http://` or `https://`.
+
+- `dst::String`: In case of `islocal=true` and `src` being a absolute path
+  `dst` defaults to `assets/<filename>`. This argument can be used to manually
+  set the location of the asset in the build directory.
 
 # Usage
 
 ```julia
 Documenter.HTML(assets = [
     # Standard local asset
     "assets/extra_styles.css",
-    # Standard remote asset (extension used to determine that class = :js)
+    # Standard remote asset (extension used to determine that class=:js)
     asset("https://example.com/jslibrary.js"),
     # Setting asset class manually, since it can't be determined manually
     asset("https://example.com/fonts", class = :css),
     # Same as above, but for a local asset
-    asset("asset/foo.script", class=:js, islocal=true),
+    asset("asset/foo.script", class = :js, islocal = true),
+    # Asset provided via source and destination path
+    asset("/abs/path/to/extra_style.css", dst = "extra_assets/extra_style.css")
 ])
 ```
 """
-function asset(uri; class = nothing, islocal=false)
+function asset(src; class = nothing, islocal = nothing, dst = nothing)
+    # automatically derive the asset class or otherwise check if it is a
+    # supported asset class
     if class === nothing
-        class = assetclass(uri)
+        class = assetclass(src)
         (class === nothing) && error("""
-        Unable to determine asset class for: $(uri)
+        Unable to determine asset class for: $(src)
         It can be set explicitly with the `class` keyword argument.
         """)
+        (class == :unknown) && error("Not supported derived class for: $(src)")
+    else
+        class in [:ico, :css, :js] || error("Unrecognised asset class $class for: $(uri)")
+    end
+    # automatically derive if the source is local or or online and perform
+    # checks if the flag has been set to true or false
+    # no file or path checks are done within this function!
+    if islocal === nothing
+        islocal = !occursin(r"^https?://", src)
+    elseif typeof(islocal)<:Bool
```

Comment:
```suggestion
    elseif isa(islocal, Bool)
```

would be more idiomatic for this check.

However, actually, if we'd constrain `islocal` to `::Union{Bool, Nothing}` in the method signature of `asset`, we could just use the inner condition directly and and drop the `else`, since we know that it has to be bool if it comes to this `else` branch.

Reply:
True `isa(islocal, Bool)` is cleaner (forgot about this one).

For the `asset()` function i can live with `islocal::Union{Bool, Nothing}` so i will change it that way.